challengeFinalComponents/PrimaryButton/BackButton.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  BackButton.swift
+//  challengeFinal
+//
+//  Created by Pietro Pugliesi on 22/10/20.
+//
+
+import UIKit
+
+public protocol BackButtonDelegate: class {
+    func buttonPressed(_ tag: Int)
+}
+
+class BackButton: UIButton {
+    public weak var delegate: BackButtonDelegate?
+    
+    public override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setupButton()
+    }
+    
+    public init(title: String, for state: UIControl.State = .normal) {
+        super.init(frame: .zero)
+        setupButton()
+        setTitle(title, for: state)
+        #warning("ou title=Voltar hard coded?")
+        addTarget(self, action: #selector(handleTap), for: .touchUpInside)
+    }
+    
+    @objc private func handleTap() {
+        self.delegate?.buttonPressed(tag)
+    }
+    
+    private func setupButton() {
+        //transparent background
+        self.backgroundColor = .none
```

Comment:
shouldn't be .clear? 
this comment is useless

Reply:
updated